### DISCUSSION TITLE
non-interactive node join in a VM image

### DISCUSSION THREAD

**Discussion by KasperSkytte** [gravitational/teleport#12516] [Q&A] [answered]
I want to create a base image for my VM's that has Teleport preinstalled and configured. Is it possible to authenticate non-interactively by setting things up beforehand in the image, so that I can simply create a VM based on the image, and it has automatically authenticated with Teleport giving access to it straight away without having to first go through an OpenSSH session and authenticate manually? I can't seem to find information about it in the docs, or which search terms to use.

**Comment by webvictim** [comment]
When you say authenticate manually, you mean `tsh login`? It isn't currently possible to script that as it either requires an SSO callback or interactive keyboard input for the password, plus 2-factor token entry etc.

If you mean you want to join the Teleport node to the cluster automatically on startup, this is definitely possible.

**Reply by KasperSkytte** [reply]
Hi Gus

Yes I mean just joining the node to the cluster automatically

**Reply by klizhentas** [reply]
@KasperSkytte is that AWS VM? If yes, you can auto join the node to cluster using AWS [IAM](https://goteleport.com/docs/setup/guides/joining-nodes-aws-iam/) joining.

**Reply by KasperSkytte** [reply]
@klizhentas unfortunately no. It's an on-premise solution. Hypervisor is openstack, and probably also proxmox

**Comment by webvictim** [comment, marked as answer]
You can set static tokens in the `/etc/teleport.yaml` config file that can be used an unlimited number of times: https://goteleport.com/docs/setup/admin/adding-nodes/#an-insecure-alternative-static-tokens

These are (as the docs suggest) less secure as anyone who gets a token can join a node to your cluster.

**Comment by KasperSkytte** [comment]
Alright. I'll avoid using long lasting tokens and just do it manually then. Just an idea though: How about new nodes could "ask" the bastion to join, and access can be granted interactively from the bastion instead, with fx `tctl`?
